packages/yoroi-extension/app/utils/recoveryPhrase.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+
+export type SortedRecoveryPhraseEntry = {|
+    id: number,
+    word: string,
+    originalIdx: number,
+|};
+
+export function makeSortedPhrase(recoveryPhrase: Array<string>): SortedRecoveryPhraseEntry[] {
+  const sorted = recoveryPhrase.slice().sort();
+  const wordIndexes = new Set();
+
+  return sorted.map((sortedWord, sortedWordIdx) => {
+    const originalIdx = recoveryPhrase.findIndex((originalWord, idx) => {
```

Comment:
@AhmedIbrahim336 , same thing here, for a recovery phrase `'b b a'`, you would get sorted array `['a', 'b', 'b']` and then the `originalIdx` for both `b` would be `0`. Never assume you have unique words. You can fix this by preserving the original index before sorting and then sort by a specific condition, e.g.:

```
const sorted = recoveryPhrase
  .map((word, originalIndex) => { word, originalIndex })
  .sort((a, b) => a.word.localeCompare(b.word))
```